tagmanager/src/tm_source_file.c

```diff
@@ -109,6 +109,35 @@ gchar *tm_get_real_path(const gchar *file_name)
 	return NULL;
 }
 
+/* add argument list of __init__() Python methods to the class tag */
+static void update_python_arglist(const TMTag *tag)
```

Comment:
But if you have a look how it's done now, it's much worse and much more unmaintainable than this - it introduces diff to the python parser compared to universal-ctags and worse, it makes a very strange direct path python_parser->TM (no other parser does this, all of them generate tags the standard way).

Such a strange callback would definitely not become part of ctags API if it's ever turned into a library so we'd have to maintain our own version. The "just one more" is very well justified in this case IMO (other option is not to add the constructor parameters to the class type and drop this feature).